tests/Doctrine/Tests/DBAL/ConnectionTest.php

```diff
@@ -403,13 +403,12 @@ public function testFetchColumn()
         $this->assertSame($result, $conn->fetchColumn($statement, $params, $column, $types));
     }
 
-    public function testConnectionIsClosed()
+    public function testGetPlatformVersionDoesNotFailOnBadConnection()
     {
-        $this->_conn->close();
-
-        $this->setExpectedException('Doctrine\\DBAL\\Exception\\DriverException');
-
-        $this->_conn->quoteIdentifier('Bug');
+        // the connection uses bogus connection requirements, so connecting
+        // *will* fail. We should be able to call this without connection
+        // and causing that exception
+        $this->_conn->getDatabasePlatform();
```

Comment:
Actually, the database platform will *never* be null, and the point of the test is simply to show that asking for the database platform (with invalid connection credentials, which this test class uses) no longer causes an exception. If you removed my "fix" in this PR, calling `getDatabasePlatform()` would actually explode.

I can add an assertion that the database platform is in fact *not* null (it is *always* determine, it's just that it uses the default platform if it cannot connect to get the version), but the real point is that this doesn't cause an exception.

I feel like you and I aren't *quite* on the same page with this change and #784. I am vastly less qualified than you in this library, so perhaps it's my mistake, and if so - I appreciate the help clarifying :). Overall, I'm simply trying to fix the change in behavior where calling `getDatabasePlatform()` began throwing an exception if a connection couldn't be made (instead of using the default platform, like in earlier versions of DBAL).

Thanks!